src/com/lms/service/AdministratorService.java

```diff
@@ -0,0 +1,195 @@
+package com.lms.service;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import com.lms.model.Author;
+import com.lms.model.Book;
+import com.lms.model.Borrower;
+import com.lms.model.Branch;
+import com.lms.model.Loan;
+import com.lms.model.Publisher;
+
+/**
+ * A service interface to ease the creation of a UI for administrative users.
+ *
+ * @author Salem Ozaki
+ * @author Jonathan Lovelace
+ */
+public interface AdministratorService extends Service {
+	/**
+	 * Create a book with the given title, author, and publisher and store it in the
+	 * database.
+	 *
+	 * @param title     the title of the book
+	 * @param author    the author of the book
+	 * @param publisher the publisher of the book
+	 * @return the newly-created book
+	 */
+	Book createBook(String title, Author author, Publisher publisher);
```

Comment:
shoudn't it be  Book createBook(String title, String authorName, String publisherName) and the service will get the author and publisher objects from the data base if they exist and if they do not the service will create the two new objects ! I don't think passing an object from the menu  to the service is a good Idea ! because no object should be created in the menu ! the menu should be just an interaction interface with the user !

Reply:
You are correct, that would not be a good idea to create a model in the menu, but that is not how I envisioned the menu to act. When the user picks an author and publisher through a list we provide them, that list will contain the author and publisher objects already, using the getAllPublisher and Author method.

Reply:
Yes. The UI that the "assignment" spec describes has the user mostly choosing items from lists; while it leaves the "administrator" UI for adding books to the database up to us, it's more consistent with the rest of the UI as it describes it to have the user choose the author and publisher from lists than to get them by name from a free-form text input.